app/src/main/java/com/bachelor_group54/funnregistrering/FragmentEnkeltFunn.java

```diff
@@ -280,6 +312,91 @@ public void updateFind() {
         funn.setFylke(fylke.getText().toString());
     }
 
+    public void pdfGenerator(){
+        // Creation of an object variable for the PDF document
+        PdfDocument pdfDocument = new PdfDocument();
+
+        //Paint is used to draw shapes and add text
+        Paint logo = new Paint();
+        Paint text = new Paint();
+
+        /*Adding pageInfo to the the PDF
+        * Passing the width, height and number of pages
+        * Creates the PDF */
+        PdfDocument.PageInfo pageInfo = new PdfDocument.PageInfo.Builder(pdfWidth,pdfHeight,1).create();
+
+        //sets the PDFs startpage.
+        PdfDocument.Page page = pdfDocument.startPage(pageInfo);
+
+        //creates canvas variable from the startpage.
+        Canvas canvas = page.getCanvas();
+
+        /*Draws image on pdf file:
+        * bitmap is the 1st parameter,
+        * position from left is the 2nd parameter,
+        * position from top is the 3rd parameter,
+        * the paint variable is the 4th parameter. */
+        canvas.drawBitmap(scalebmp,56,40,logo);
+
+        // adding typeface for the text
+        text.setTypeface(Typeface.create(Typeface.DEFAULT,Typeface.NORMAL));
+
+        //Setting text size (in the pdf)
+        text.setTextSize(16);
+
+        /* Draws the text on the pdf
+        * the text is the 1st parameter,
+        * the position from start is the 2nd parameter,
+        * the position from top is the 3rd parameter,
+        * the paint variable  (text) is the 4th parameter.
+        * */
+        canvas.drawText("Dette er en test for å se om alt fungerer som det skal", 209,100, text);
+        canvas.drawText("Her er enda en test", 209,80, text);
+        canvas.drawText("Og dette er enda litt mer testing", 209,60, text);
+
+        //finishing the page
+        pdfDocument.finishPage(page);
+
+        //sets storage path
+        File file = new File(Environment.getExternalStorageDirectory(),"pdfTest.pdf");
```

Comment:
Ikke sikert du trenger noen permissions om du lagrer i appens filer.
Eksempel kode fra ImageSaver
String path =  context.getFilesDir().getPath(); //Gets the path to the program
String filename = "/Image-" + pictureNr +".jpg"; //Sets the iamge name
File file = new File (path + filename); // Combines the program path and the filename